README.md

```diff
@@ -194,6 +194,27 @@ PasswordEncrypted: '@@FILE:/dir/@@PROP:name@@.txt@@'
 PasswordEncrypted: '@@FILE:@@ORACLE_HOME@@/dir/name.txt@@'
 ```
 
+**Environment tokens** are declared with the syntax `@@ENV:<name>@@`. This type of token is resolved by looking up the system environment variable `<name>`, and substituting that value for the token.
+
+**Secret tokens** are declared with the syntax `@@SECRET:<name>:<key>@@`. This type of token is resolved by determining the location of a Kubernetes secret file, and reading the first line from that file. That line is substituted for the token.
+
+There are two methods for deriving the location of the Kubernetes secret file. The first method involves using one or more configured root directories, and looking for the secret file in the paths `<root-directory>/<name>/<key>`.
+
+The root directories are configured as a comma-separated list of directories, using the environment variable `WDT_MODEL_SECRETS_DIRS`. For example, `WDT_MODEL_SECRETS_DIRS` is set to `/etc/my-secrets,/etc/your-secrets`, the token `@@SECRET:secrets:the-secret@@` will search the following locations:
+```
+/etc/my-secrets/secrets/the-secret
+/etc/your-secrets/secrets/the-secret
+``` 
+If either of these files is found, the secret is read from that file and substituted in the model.
+
+The second method for locating the Kubernetes secret file is to use the environment variable `WDT_MODEL_SECRETS_NAME_DIR_PAIRS` to map `<name>` values to specific directory locations. For example, if `WDT_MODEL_SECRETS_NAME_DIR_PAIRS` is set to `my-root=/etc/my-secrets,your-root=/etc/your-secrets`, the token `@@SECRET:your-root:the-secret@@` will look for the secrets file at: 
+```
+/etc/your-secrets/the-secret
+```
+If the `<name>` value has a corresponding mapped directory in `WDT_MODEL_SECRETS_NAME_DIR_PAIRS`, that directory will take precedence over any roots specified in `WDT_MODEL_SECRETS_DIRS`. 
+
+NOTE: it is important that the secrets directories contain only secrets files, since those files are examined to create a list of available name/key pairs.  
```

Comment:
NOTE: it is important -> NOTE: It is important
since those files -> because those files